src/derived.rs

```diff
@@ -358,8 +358,13 @@ where
         if let Some(old_memo) = &old_memo {
             if let Some(old_value) = &old_memo.value {
                 if MP::memoized_value_eq(&old_value, &result.value) {
-                    assert!(old_memo.changed_at <= result.changed_at.revision);
-                    result.changed_at.revision = old_memo.changed_at;
+                    match (&old_memo.inputs, &result.subqueries) {
+                        (MemoInputs::Untracked, _) | (_, None) => (),
```

Comment:
I'm not convinced this is a bug. Indeed, the value **did not** change since that revision, so re-running downstream stuff should not produce any new result.

I think the important thing is that the `inputs` still contain `Untracked`, and the `verified_at` field should still be out of date come the next revision.

Since the `verified_at` is old, we should never upgrade the entry to the current revision, so it should always be recomputed if probed...

Reply:
Yeah: as I said, this doesn't actually fix the problem I am observing. Still I am unsure about the meaning of `changed_at` field in the presence of untracked input. Perhaps we need to use an `enum` such that you have `changed_at` only if you are a tracked non-constant?